flypipe/node.py

```diff
@@ -344,6 +365,11 @@ def _run_sequential(self, spark, execution_graph, outputs):
                     ),
                 )
 
+                # If cache exists, and the transformation has run, then save its cache
+                runnable_node["node_run_context"].cache_context.write(

```

Comment:
Would be better if we can avoid duplicating the cache write call for sequential and parallel processing.

Reply:
problem is that they shall be done after noderesult